source/common/ssl/context_config_impl.cc

```diff
@@ -1,17 +1,43 @@
 #include "common/ssl/context_config_impl.h"
 
+#include <memory>
 #include <string>
 
 #include "common/common/assert.h"
 #include "common/config/datasource.h"
 #include "common/config/tls_context_json.h"
 #include "common/protobuf/utility.h"
+#include "common/ssl/tls_certificate_config_impl.h"
 
 #include "openssl/ssl.h"
 
 namespace Envoy {
 namespace Ssl {
 
+namespace {
+
+std::string readConfig(
+    const envoy::api::v2::auth::CommonTlsContext& config, Secret::SecretManager& secret_manager,
+    const std::function<std::string(const envoy::api::v2::auth::TlsCertificate& tls_certificate)>&
+        read_inline_config,
+    const std::function<std::string(const std::shared_ptr<Ssl::TlsCertificateConfigImpl>& secret)>&
```

Comment:
the `std::function` doesn't have to take shared_ptr, just use `const Ssl::TlsCertificateConfigImpl&`